python_modules/dagster/dagster/_core/definitions/run_request.py

```diff
@@ -72,6 +73,9 @@ class RunRequest(
             Required for sensors that target multiple jobs.
         asset_selection (Optional[Sequence[AssetKey]]): A sequence of AssetKeys that should be
             launched with this run.
+        stale_assets_only (Optional[Sequence[AssetKey]]): Set to true to further narrow the asset
+            selection to stale assets. If passed without an asset selection, all stale assets in the
```

Comment:
Responding to the second sentence, I would expect this parameter to only work in situations where either the sensor, the run request, or the job are already targeting an asset selection.  If none of those are targeting an asset selection, I think we should error.  I.e. I think supplying this parameter should only ever reduce the set of assets that get materialized.